ptero_workflow/implementation/backend.py

```diff
@@ -65,6 +70,21 @@ def get_workflow_outputs(self, workflow_id):
         workflow = self.session.query(models.Workflow).get(workflow_id)
         return workflow.get_outputs()
 
+    def get_execution_inputs(self, execution_id):
+        execution = self.session.query(models.Execution).get(execution_id)
+        return execution.get_inputs()
+
+    def set_execution_outputs(self, execution_id, outputs):
+        execution = self.session.query(models.Execution).get(execution_id)
+        execution.set_outputs(outputs)
+        try:
+            self.session.commit()
+        except IntegrityError:
+            self.session.rollback()
+            LOG.exception('Execution %d has already had its outputs set' %
+                    execution.id)
+            raise OutputsAlreadySet
```

Comment:
Determining when outputs match is tricky business.  A naive approach would simply determine whether outputs match, byte-for-byte, but byte-for-byte comparison often not useful since it is possible for two different data structures to be slightly different yet be equivalent for all practical purposes.

When outputs genuinely do not match (ie, one of the results are wrong), that indicates a genuine problem.  It must be determined which output(s) are invalid, and probably requires manual investigation.